### PR TITLE
Removed unnecessary warning for 'outputs' parameter

### LINKED ISSUE #25909: Remove unnecessary postprocessor outputs warning
## Reason
The following warning is given, even when `outputs = none`:
```
A Postprocessor utilizes the 'outputs' parameter; however, postprocessor output is disabled for the 'console' output object.
```
This occurs in `AdvancedOutput::initPostprocessorOrVectorPostprocessorLists`.

## Design
This would fix the logic to avoid this warning when it is not needed.

## Impact
Less false warnings.

### PR DESCRIPTION
**Pull request by joshuahansel**
Closes #25909

Also, cleaned up some grammar in the other warning in this method.

### CODE REVIEW

**Comment by joshuahansel** [comment]
Just checking which tests fail first.

File: framework/include/outputs/AdvancedOutput.h [hunk 1]
```diff
@@ -412,35 +409,19 @@ AdvancedOutput::initPostprocessorOrVectorPostprocessorLists(const std::string &
...
-      if (execute_on_name == "execute_postprocessors_on")
-        mooseWarning("A Postprocessor utilizes the 'outputs' parameter; however, postprocessor "
-                     "output is disabled for the '",
-                     name(),
-                     "' output object.");
-      else if (execute_on_name == "execute_vectorpostprocessors_on")
-        mooseWarning("A VectorPostprocessor utilizes the 'outputs' parameter; however, vector "
-                     "postprocessor output is disabled for the '",
+      {
+        const bool is_pp_type = (execute_data_name == "postprocessors") ? true : false;
```
Comment (lindsayad) [inline framework/include/outputs/AdvancedOutput.h:413]:
```suggestion
        const bool is_pp_type = (execute_data_name == "postprocessors");
```

File: framework/src/outputs/AdvancedOutput.C [hunk 2]
```diff
@@ -150,6 +150,10 @@ AdvancedOutput::initialSetup()
 void
 AdvancedOutput::init()
 {
+  // Initialize the execution flags
+  for (auto & it : _advanced_execute_on)
+    initExecutionTypes(it.first, it.second);
```
Comment (lindsayad) [inline framework/src/outputs/AdvancedOutput.C:155]:
```suggestion
  for (auto & [name, input] : _advanced_execute_on)
    initExecutionTypes(name, input);
```

File: test/tests/outputs/format/tests [hunk 3]
```diff
@@ -109,8 +109,7 @@
   [pps_screen_out_warn_test]
     type = 'RunApp'
     input = 'pps_screen_out_warn.i'
-    expect_out = "A Postprocessor utilizes the 'outputs' parameter; however, postprocessor output is "
-                 "disabled for the 'console' output object."
+    expect_out = "The post-processor 'avg_block' has requested to be output by the 'console' output, but post-processor output is disabled for this output object"
```
Comment (lindsayad) [inline test/tests/outputs/format/tests:112]: I think "that" would read better than "this"

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/include/outputs/AdvancedOutput.h (hunk 1 above)
Comment (joshuahansel) [reply framework/include/outputs/AdvancedOutput.h:413]: 🤦

**Review by lindsayad — APPROVED** [review]

**Pull request merged**